### ISSUE TITLE
[BUG] [ADL][ZEPHYR] failed to enable target core for widget SSP0.OUT on ADLP ZEPHYR platform

### ISSUE THREAD

**Issue by keqiaozhang** [thesofproject/sof#5018]
**Describe the bug**
When playing audio on adlp zephyr platform, it runs to:
```
[  126.590714] kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: ipc tx error for 0x40070000 (msg/reply size: 12/12): -22
[  126.590740] kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: error: failed to enable target core for widget SSP0.OUT
[  126.590750] kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: error: failed setting up DAI widget SSP0.OUT
[  126.590761] kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: ASoC: error at snd_soc_dai_hw_params on SSP0 Pin: -22
[  126.590780] kernel:  NoCodec-0: ASoC: soc_pcm_hw_params() failed (-22)
[  126.590792] kernel:  Port0: ASoC: dpcm_fe_dai_hw_params failed (-22)
```
After debugging, this regression is related to recently tplg changes(multi-core enabling).

**To Reproduce**
aplay   -Dhw:0,0 -r 48000 -c 2 -f S16_LE /dev/zero -vvv

**Reproduction Rate**
100%
**Expected behavior**
no ipc error

**dmesg**
```
[  126.590315] kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: DAI config 0 matches pcm hw params
[  126.590324] kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: DAI config 0 matches pcm hw params
[  126.590327] kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: rate_min: 48000 rate_max: 48000
[  126.590332] kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: channels_min: 2 channels_max: 2
[  126.590343] kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: DAI config 0 matches pcm hw params
[  126.590346] kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: DAI config 0 matches pcm hw params
[  126.590350] kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: rate_min: 48000 rate_max: 48000
[  126.590354] kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: channels_min: 2 channels_max: 2
[  126.590365] kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: ipc tx: 0x40070000: GLB_PM_MSG: CORE_ENABLE
[  126.590714] kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: ipc tx error for 0x40070000 (msg/reply size: 12/12): -22
[  126.590740] kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: error: failed to enable target core for widget SSP0.OUT
[  126.590750] kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: error: failed setting up DAI widget SSP0.OUT
[  126.590761] kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: ASoC: error at snd_soc_dai_hw_params on SSP0 Pin: -22
[  126.590780] kernel:  NoCodec-0: ASoC: soc_pcm_hw_params() failed (-22)
[  126.590792] kernel:  Port0: ASoC: dpcm_fe_dai_hw_params failed (-22)
[  126.590809] kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: pcm: free stream 0 dir 0
[  126.591531] kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: pcm: close stream 0 dir 0
[  126.602977] kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: ipc rx: 0x90020000: GLB_TRACE_MSG: DMA_POSITION
[  126.603031] kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: ipc rx done: 0x90020000: GLB_TRACE_MSG: DMA_POSITION
[  126.603949] kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: ipc rx: 0x90020000: GLB_TRACE_MSG: DMA_POSITION
[  126.603972] kernel: sof-audio-pci-intel-tgl 0000:00:1f.3: ipc rx done: 0x90020000: GLB_TRACE_MSG: DMA_POSITION
```

**Environment**
1) Kernel Branch: topic/sof-dev /a9c617ef
2) SOF Branch: main /cd48b895ebda
3) Zephyr Commit: 73a957e4b316
4) Topology: sof-adlp-nocodec.tplg
5) Platform: adlp-zephyr

**Comment by lyakh** [comment]
Zephyr is currently limited to using 2 cores on Intel ADSP, this should be eliminated when https://github.com/zephyrproject-rtos/zephyr/pull/39603 is merged

**Comment by XiaoyunWu6666** [comment]
FYI 
in inner daily test 8518 
on  model ADLP_RVP_NOCODEC_ZEPHYR
in testcase check-alsabat-headset-playback
still failed to enable target core for widget SSP0.OUT

**Comment by lgirdwood** [comment]
@kv2019i I assume we are bringing up the target core after core 0 is fully booted. @andyross is asynchronous secondary core booting/power off supported today or still WIP for some platforms ?

**Comment by kv2019i** [comment]
There's first a simple problem that the core count default is 2 on Zephyr side. That is easy to fix. But with that done, we are seeing some weird errors. With a topology with four PCM pipelines assigned to cores 0, 1, 2 and 3, tests works ok on cores 0 and 3, but the secondary boot fails on cores 1 and 2. Debugging continues. This should be supported in both Zephyr and SOF.

**Comment by kv2019i** [comment]
Not a complete fix, but gets us further along: https://github.com/kv2019i/zephyr/pull/new/topic/cavs25-4cores-default

**Comment by kv2019i** [comment]
@lgirdwood More issues found, maybe we need to move this to v2.1? New issue https://github.com/thesofproject/sof/issues/5050

**Comment by kv2019i** [comment]
Depends on Zephyr https://github.com/zephyrproject-rtos/zephyr/pull/41764

**Comment by lgirdwood** [comment]
Duplicate with #5432

**Comment by keqiaozhang** [comment]
This issue is blocked by https://github.com/zephyrproject-rtos/zephyr/issues/43087.

**Comment by kv2019i** [comment]
To fix this issue, https://github.com/zephyrproject-rtos/zephyr/pull/41764 is mandatory but not enough on its own. Steps to fix stil needed:
- get above Zephyr fix to SOF (gated by https://github.com/zephyrproject-rtos/zephyr/issues/43087 )
- change SOF to use a different target for TGLLP/TGLH builds (cannot be done before above or otherwise we break all TGLH targets)
- change core count for TGL-LP Zephyr in mainline
- verify above fix with SOF (TGL-LP nocodec tests should now pass in SOF CI)

**Comment by kv2019i** [comment]
@keqiaozhang  We can close this now, right?
